src/main/java/net/sf/jabref/logic/search/rules/GrammarBasedSearchRule.java

```diff
@@ -146,10 +148,28 @@ public boolean compare(BibEntry entry) {
                 return matchFieldValue(entry.getType());
             }
 
+            // special case for searching a single keyword
+            if (fieldPattern.matcher("keyword").matches()) {
+                if (entry.hasField(FieldName.KEYWORDS)) {
+                    List<String> keywords = new ArrayList<>(entry.getKeywords());
```

Comment:
And you could even use that together with the prev line if you use your getFieldOptional method